mpi-proxy-split/lower-half/lower-half.cpp

```diff
@@ -833,14 +831,16 @@ static int restoreMemoryArea(int fd, MtcpHeader *ckptHdr)
   ProcMapsArea area;
 
   int rc = read(fd, &area, sizeof area);
+  assert(rc == sizeof area);
+  JASSERT(mprotect(area.addr, area.size, area.prot) == 0);
```

Comment:
At this point, MANA is reading the "area" structure from the checkpoint image file. The area is not in the memory yet (mapped at line 939 in this version). So calling protect on `area.addr `will fail.